### ISSUE TITLE
[Bug]: Update to Groupfolders 18.0.4 has changed OCS response structure for /apps/groupfolders/folders

### ISSUE THREAD

**Issue by jngrb** [nextcloud/groupfolders#3414]
### Steps to reproduce
1. Have a NC installation with Groupfolders v18.0.3 (in our case NC 30.0.1)
2. Use /apps/groupfolders/folders OCS REST API
3. Upgrade to Groupfolders v18.0.4
4. Use /apps/groupfolders/folders OCS REST API
 
### Expected behaviour
No change to OCS response structure as it was a patch level upgrade

### Actual behaviour
OCS response structure changed from a dictionary
{"1" -> entry1, "2" -> entry2, ...}
to a simple array / list
[entry1, entry2]

### Server configuration

**Operating system:** Debian 12 Bookworm

**Web server:** Apache/2.4.62

**Database:** tested with sqlite3

**PHP version:** v8.2.25

**Nextcloud version:** v30.0.1

**Group folders version:** v18.0.4

**Updated from an older Nextcloud/ownCloud or fresh install:** Upgrade of Groupfolders from previous version

### Detailed example output

Old (Groupfolders v18.0.4)

```[json]
{
  "ocs": {
    "meta": {
      "status": "ok",
      "statuscode": 100,
      "message": "OK",
      "totalitems": "",
      "itemsperpage": ""
    },
    "data": {
      "1": {
        "id": 1,
        "mount_point": "/dc8f91a7-0630-423c-99dc-423392fa3ad5",
        "groups": [],
        "quota": 1024,
        "size": 0,
        "acl": false,
        "manage": [],
        "group_details": []
      },
      "123": { <...> },
      <...>
    ]
  }
}
```

New (Groupfolders v18.0.4)

```[json]
{
  "ocs": {
    "meta": {
      "status": "ok",
      "statuscode": 100,
      "message": "OK",
      "totalitems": "",
      "itemsperpage": ""
    },
    "data": [
      {
        "id": 1,
        "mount_point": "/dc8f91a7-0630-423c-99dc-423392fa3ad5",
        "groups": [],
        "quota": 1024,
        "size": 0,
        "acl": false,
        "manage": [],
        "group_details": []
      },
      {
        "id": 123,
        <...>
      }
      <...>
    ]
  }
}
```

**Comment by provokateurin** [comment]
Thanks for the report, I'll fix it right away!

**Comment by provokateurin** [comment]
Fixed in Groupfolders 18.0.5
